crypto/crypto/src/ed25519.rs

```diff
@@ -410,6 +431,7 @@ impl Signature for Ed25519Signature {
         message: &T,
         public_key: &Ed25519PublicKey,
     ) -> Result<()> {
+        precondition!(has_tag!(public_key, ValidPublicKeyTag)); // Public keys should be valid.
```

Comment:
The comment does not add any value beyond what the annotation already provides.

Reply:
I for one appreciate these kind of comments : o (I also wouldn't mind the comment being on its own line)